src/hooks/useBrowserTabVisibilityChange.ts

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+
+export function useBrowserTabVisibilityChange() {
+    const [isBrowserTabVisible, setIsBrowserTabVisible] = React.useState(true);
+    const handleFocusHandleVisibilityChange = () => {
+        if (document.hidden) {
+            setIsBrowserTabVisible(false);
+        } else {
+            setTimeout(() => setIsBrowserTabVisible(true), 250);
+        }
+    };
+
+    React.useEffect(() => {
+        document.addEventListener('visibilitychange', handleFocusHandleVisibilityChange);
+
+        return () => {
+            document.removeEventListener('visibilitychange', handleFocusHandleVisibilityChange);
+        };
+    });
+
+    return isBrowserTabVisible;
```

Comment:
It makes sense to wrap all code of the hook to `useEffect` with the second argument `[]`.
Because now every rerender of the component add `addEventListener` to the document. It should be added just once on the init state.